src/backend/src/services/projects.services.ts

```diff
@@ -605,6 +606,58 @@ export default class ProjectsService {
   }
 
   /**
+   * Create an assembly
+   * @param name The name of the assembly to be created
+   * @param pdmFileName The name of the file holding the assembly
+   * @param userCreated The user creating the assembly
+   * @param wbsElementId The
+   * @returns the project that the user has favorited/unfavorited
+   * @throws if the project wbs doesn't exist or is not corresponding to a project
+   */
+  static async createAssembly(
+    name: string,
+    userCreated: User,
+    wbsNumber: WbsNumber,
+    pdmFileName?: string
+  ): Promise<Assembly> {
+    if (!isProject(wbsNumber)) throw new HttpException(400, `${wbsPipe(wbsNumber)} is not a valid project WBS #!`);
+    const { carNumber, projectNumber, workPackageNumber } = wbsNumber;
+
+    const project = await prisma.project.findFirst({
+      where: {
+        wbsElement: {
+          carNumber,
+          projectNumber,
+          workPackageNumber
+        }
+      },
+      ...projectQueryArgs
+    });
+
+    if (!project) throw new NotFoundException('Project', wbsPipe(wbsNumber));
+    if (project.wbsElement.dateDeleted) throw new DeletedException('Project', project.projectId);
+
+    const { teams, wbsElementId } = project;
+
+    if (!isAdmin(userCreated.role) && !isUserPartOfTeams(teams, userCreated))
+      throw new AccessDeniedException('Users must be admin, or assigned to the team to create assemblies');
+
+    const userCreatedId = userCreated.userId;
+
+    const assembly = await prisma.assembly.create({
+      data: {
+        name,
```

Comment:
name is a unique field, so you should add a check to make sure there isn't already an assembly with the specified name and throw and Httpexception if there is